apps/linkedin-to-notion/src/contents/linkedin-notion-side-panel.tsx

```diff
@@ -42,6 +49,11 @@ const LinkedinNotionSidePanel = () => {
       }
       if (data.session) {
         setUser(data.session.user);
+        // There's one thing we're specifically interested in in the data.session object
+        // It's the provider_token (i.e notion's api access token in this case)
+        // This token is necessary to make api calls to notion elsewhere in the app
+        // There's no way to retrieve this token if we don't save it here
+        storeEntriesSecurely('authData', data.session);
```

Comment:
I may be missing something but this is already saved in a key called user using the hook `useStorage`. Am I missing something?

Reply:
lmao you're absolutely right 😵 except that instead of storing `data.session.user` only, we should store `data.session` altogether (bc Notion's API token is the root). Here's the piece of code you're referring to in `linkedin-notion-side-panel.tsx`:

```js
const LinkedinNotionSidePanel = () => {
  const [isOpen, setIsOpen] = useState(true); // set-back to false before deployment
  const [user, setUser] = useStorage<User>({
    key: 'user',
    instance: new Storage({
      area: 'local',
    }),
  });

  useEffect(() => {
    async function init() {
      const { data, error } = await supabase.auth.getSession();

      if (error) {
        console.error(error);
        return;
      }
      if (data.session) {
        setUser(data.session.user); // 👈 Let's store the whole session here, not only the user
        sendToBackground({
          name: 'sessions/resolvers/init-session',
          body: {
            refresh_token: data.session.refresh_token,
            access_token: data.session.access_token,
          },
        });
      }
    }

    init();
  }, []);
```

Let's forget about secure storage too for now!